src/transformers/modeling_t5.py

```diff
@@ -872,6 +1017,11 @@ def __init__(self, config):
     def get_input_embeddings(self):
         return self.shared
 
+    def set_output_past(self, do_output_past: bool):
+        self.config.output_past = do_output_past
+        self.decoder.output_past = do_output_past
+        self.encoder.output_past = do_output_past
```

Comment:
Same comment as in the parent class, the output_past variables in the decoder blocks and attention layer need to be set.

Reply:
Thanks for catching that @yjernite! Looking at this function this also makes me wonder whether it might be better to use a `use_cache` boolean flag argument instead of a `config.output_past` flag (which @sshleifer proposed and which is implemented in https://github.com/patrickvonplaten/transformers/blob/e20a76f52a40300357b6e1f70a3e896d39840095/src/transformers/modeling_bart.py#L414).

I guess the Pro's and Con's are:

Pros for `use_cache` / Cons for `output_past`:
- more control for the user as he decides at function call whether to output past or not
- seems logical to be correct to use it here, because it is not really an attribute of the model
- some models can't output the past or it doesn't make sense (models that don't use a causal mask)
- less under-the-hood workings and easier to see for the user what's going on

Cons for `use_cache` / Pros for `output_past`:
- we would have to change a lot of code in GPT2, GPT, CTRL, T5 and their docstrings
- might be a breaking change if we remove the `output_past` from the configs

I'm in favor of using `use_cache` as @sshleifer is I think. What is your opinion on that @LysandreJik, @thomwolf, @yjernite, @sshleifer?

Reply:
@sshleifer and I could do the changes in a separate PR and we could default the `use_cache` to True so there shouldn't be breaking changes.

Reply:
Definitely for. I had something similar in my local branch for a bit. Mostly I think it's safer / will lead to fewer silent bugs.

Reply:
Also one thing to consider is that some memory efficient models would like to switch `use_cache` off to be able to run on long sequences, so it's not always better to use the cache.

Reply:
I'm ok with using `use_cache`. I just don't want this to be a breaking change if possible.